src/distilabel/steps/task/evol_instruct/base.py

```diff
@@ -0,0 +1,213 @@
+        if not self.output_mappings:
+            self.output_mappings = {k: k for k in self.outputs}
+
+        _output = {}
+        if not self.store_evolutions:
+            _output[self.output_mappings["evolved_instruction"]] = instructions[-1]
+        else:
+            _output[self.output_mappings["evolved_instructions"]] = instructions
+
+        if self.generate_answers and answer is not None:
+            _output[self.output_mappings["answer"]] = answer
+
+        _output[self.output_mappings["model_name"]] = self.llm.model_name
+        return _output
+
+    @override
+    def process(self, inputs: "StepInput") -> "StepOutput":  # type: ignore
+        """Processes the inputs of the task and generates the outputs using the LLM.
+
+        Args:
+            inputs: A list of Python dictionaries with the inputs of the task.
+
+        Returns:
+            A list of Python dictionaries with the outputs of the task.
+        """
+        enum_attributes = [
+            member.name  # type: ignore
+            for member in self.mutation_templates.__members__.values()  # type: ignore
+        ]
+
+        instructions: List[List[str]] = [[input["instruction"]] for input in inputs]
+        for iter_no in range(self.num_evolutions):
+            formatted_prompts = []
+            for instruction in instructions:
+                mutation = np.random.choice(enum_attributes)
+                formatted_prompts.append(
+                    self.mutation_templates[mutation].value.replace(  # type: ignore
+                        "<PROMPT>", instruction[-1]
+                    )
+                )
+
+            formatted_prompts = [
+                self.format_input(prompt) for prompt in formatted_prompts
+            ]
+            generated_prompts = self.llm.generate(
+                formatted_prompts,
+                **self.generation_kwargs,  # type: ignore
+            )
+
+            evolved_instructions = []
+            for generated_prompt in generated_prompts:
+                evolved_instructions.append(
+                    generated_prompt.split("Prompt#:")[-1].strip()
+                )
+
+            if self.store_evolutions:
+                instructions = [
+                    instruction + [evolved_instruction]
+                    for instruction, evolved_instruction in zip(
+                        instructions, evolved_instructions
+                    )
+                ]
+            else:
+                instructions = [
+                    [evolved_instruction]
+                    for evolved_instruction in evolved_instructions
+                ]
+
+            self._logger.info(
+                f"🔄 Ran iteration {iter_no} evolving {len(instructions)} instructions!"
+            )
+
+        if self.store_evolutions:
+            # Remove the input instruction from the `evolved_instructions` list
+            instructions = [instruction[1:] for instruction in instructions]
```

Comment:
I think for downstream steps using the output of this one, it would be easier to also include the original instruction.

Reply:
Yes, note that the initial instruction is already part of the input i.e. `instructions`, all those will be included within `evolved_instructions`, so the starting instruction won't get lost unless explicitly overwritten.